src/index/index.cc

```diff
@@ -116,6 +127,16 @@ Index<T>::AnnIterator(const DataSet& dataset, const Json& json, const BitsetView
     if (status != Status::success) {
         return expected<std::vector<std::shared_ptr<IndexNode::iterator>>>::Err(status, msg);
     }
+    // when index is immutable, bitset size should always equal to data count in index
+    // when index is mutable, it could happen that data count larger than bitset size, see
+    // https://github.com/zilliztech/knowhere/issues/70
+    // so something must be wrong at caller side when passed bitset size larger than data count
+    if (bitset_.size() > this->Count()) {
+        msg = fmt::format("bitset size invalid! bitset size: {}, data count: {}", bitset_.size(), this->Count());
```

Comment:
nit: I'd write in the error message that the bitset size is expected to be <= than the data size, just to make the error more clear. Otherwise, I'd assume (based on the current message) that these two sizes must be equal.